source/extensions/priv/server/ntds.c

```diff
@@ -0,0 +1,195 @@
+#include "precomp.h"
+
+/*! @brief Typedef for the NTDSContext struct. */
+typedef struct
+{
+	struct jetState *ntdsState;
+	struct ntdsColumns *accountColumns;
+	struct decryptedPEK *pekDecrypted;
+} NTDSContext;
+
+// This is the raw NTDS command function. When the remote user
+// sends a command request for priv_ntds_parse, this function fires.
+// It calls the setup routines for our Jet Instance, attaches the isntance
+// to the NTDS.dit database the user specified, and creates our channel.
+// The user interacts with the NTDS database through that channel from that point on.
+DWORD ntds_parse(Remote *remote, Packet *packet){
+	Packet *response = packet_create_response(packet);
+	DWORD res = ERROR_SUCCESS;
+	struct jetState *ntdsState = calloc(1,sizeof(struct jetState));

```

Comment:
engine_shutdown should only get called if the initial setup fails or if the user calls close channel. the engine stays open between calls because of the channelised approach